src/Nethermind/Nethermind.TxPool.Test/TxPoolTests.Blobs.cs

```diff
@@ -132,17 +132,22 @@ public void should_reject_txs_with_nonce_too_far_in_future(TxType txType, int ma
             };
             _txPool = CreatePool(txPoolConfig, GetCancunSpecProvider());
             EnsureSenderBalance(TestItem.AddressA, UInt256.MaxValue);
-            for (int nonce = 0; nonce < txPoolConfig.Size; nonce++)
+
+            Transaction[] txs = new Transaction[txPoolConfig.Size];
+            Parallel.For(0, txPoolConfig.Size, (nonce) =>
             {
-                Transaction tx = Build.A.Transaction
+                txs[nonce] = Build.A.Transaction
                     .WithNonce((UInt256)nonce)
                     .WithType(txType)
                     .WithShardBlobTxTypeAndFieldsIfBlobTx()
                     .WithMaxFeePerGas(1.GWei())
                     .WithMaxPriorityFeePerGas(1.GWei())
                     .SignedAndResolved(_ethereumEcdsa, TestItem.PrivateKeyA).TestObject;
+            });
 
-                _txPool.SubmitTx(tx, TxHandlingOptions.None).Should().Be(nonce > expectedNumberOfAcceptedTxs
+            for (int nonce = 0; nonce < txPoolConfig.Size; nonce++)
+            {
+                _txPool.SubmitTx(txs[nonce], TxHandlingOptions.None).Should().Be(nonce > expectedNumberOfAcceptedTxs
```

Comment:
`TxPool.SubmitTx` should be thread safe, the behavior might diverge though if there is a big gap in nonces though. So potentially it could be parallelized with some batching?

Reply:
You can also reduce TxPool size in first 2 cases - we don't really need 100 if it's slow. 10 should be enough

Reply:
~~In this test we are accepting first N (N = TxPool.Size) transactions and returning `AcceptTxResult.NonceTooFarInFuture` for further ones. If we parallelize it, we can accept N+1 and higher nonces and later evict when lower nonces arrive. So we can stick with non-parallelized solution or refactor test to not check acceptance at the moment of adding, but to verify the highest nonce in the pool at the very end~~

Reply:
Ah "big gap in nonces" is configured by `MaxPendingTxsPerSender` and `MaxPendingBlobTxsPerSender`. We are accepting current nonce + `MaxPending...` as the maximal nonce. It is configured at the test level so actually we should reject N+1 nonces and accept only up to N. So should be parallelizable even without refactoring `AcceptTxResult.NonceTooFarInFuture`

Reply:
But for blob txs we are not accepting ANY gaps so adding must be consecutive